Add missing docs for COLLECT (#605)
neo4j/docs-cypher: modules/ROOT/pages/keyword-glossary.adoc

> | xref::syntax/expressions.adoc#query-syntax-case[CASE Expression]

Reviewer: ```suggestion
| xref::syntax/expressions.adoc#query-syntax-case[CASE]
```

| A generic conditional expression, similar to if/else statements available in other languages.

| xref:syntax/expressions.adoc#collect-subqueries[COLLECT {...}]

> | An expression used to create a list with the rows returned by a subquery.

Reviewer: ```suggestion
| Creates a list with the rows returned by a subquery.

| xref:syntax/expressions.adoc#count-subqueries[COUNT {...}]

> | An expression used to compute the number of results of a subquery.

Reviewer: ```suggestion
| Computes the number of results of a subquery.
```

| xref:syntax/expressions.adoc#existential-subqueries[EXISTS {...}]

> | An `EXISTS` expression is used to evaluate the existence of a subquery.

Reviewer: ```suggestion
| Evaluates the existence of a subquery.
```
